File: D1L1/LameKnight.cpp
#include <vector>
#include <string>
#include <cmath>
using namespace std;

int maxCells(int h, int w) {
    if (h == 1 || w == 1) return 1;
    if (h == 2) return min(4, (w + 1)/2);
    if (w >= 7) return w - 2;
    return min(4, w);
}

File: D1L1/LameKnight.java
public class LameKnight {
    public int maxCells(int height,
                        int width) {
        if (height == 1 || width == 1) {
            return 1;
        }
        if (height == 2) {
            return Math.min(4, (width + 1)/2);
        }
        if (width >= 7) {
            return width - 2;            
        }
        return Math.min(4, width);
    }
}